### PR TITLE
Mining Cyborg Ash Storm Plating Upgrade

### PR DESCRIPTION
**Pull request by GunHog**
- Adds an upgrade to Mining Borgies which will make them immune to ash
  storms.
- Upgrading a mining cyborg in this manner will give it a Lavaland themed paint job!

I want this to be a medium tech level requirement for when the borg or miners complete their first mineral trip. With dedication and a share of plasma from Xenobio or Engineering, it should be possible to scrape up enough plasma to afford it as well.

Sprites by @WJohn!
Suggestion for this PR by @KorPhaeron on https://github.com/tgstation/tgstation/pull/18279#issuecomment-223837989

:cl:
Name: Gun Hog
rscadd: Nanotrasen scientists have completed a design for adapting mining cyborgs to the Lavaland Wastes in the form of anti-ash storm plating. Research for this technology must be adapted at your local station.
/:cl:

**EDIT:** Merge token: https://github.com/tgstation/tgstation/pull/18391

screenshot 2016-06-07 10 46 23

screenshot 2016-06-07 10 46 27

### CODE REVIEW

**Comment by Armhulen** [comment]
Cool as shit 👍

**Comment by Durkel** [comment]
Very nice, they've needed this for awhile/

**Comment by GunHog** [comment]
I just **love** merge conflicts with myself!

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -658,6 +658,16 @@
 	construction_time = 120
 	category = list("Cyborg Upgrade Modules")
 
+/datum/design/borg_upgrade_ashplating
+	name = "Cyborg Upgrade (Ash Storm Plating)"
+	id = "borg_upgrade_ashplating"
+	build_type = MECHFAB
+	build_path = /obj/item/borg/upgrade/ashplating
+	req_tech = list("plasmatech" = 4, "materials" = 4, "engineering" = 4)
```
Comment (KorPhaeron) [inline code/modules/research/designs/mechfabricator_designs.dm]: @LatD 

Does this look okay to you

Comment (GunHog) [inline code/modules/research/designs/mechfabricator_designs.dm]: This is meant to be a **medium** effort item. Basically along the lines of "If Scientists are competent, they can make this item on request". I fully intend this to be something you could get by collecting plasma from xenobio or engineering.

This line of my mining related additions (satchel of holding, the corresponding borg version, this item) are intended to be obtainable in the early stages of the round if you specifically seek them out.

**Pull request merged**